### ISSUE TITLE
"encrypted" and "kms_key_id" are unsupported arguments in "root_block_device"

### ISSUE THREAD

**Issue by louis-chargin** [terraform-providers/terraform-provider-aws#9872]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.12.7
+ provider.aws v2.17.0

### Affected Resource(s)

* aws_instance

### Terraform Configuration Files

main.tf
```hcl
module "ec2" {
  source = "./modules/ec2"
  ec2_type = "${var.ec2_type}"
  region = "${local.region}"
  ssh_key_name = "${var.ssh_key_name}"
  private_subnet_zoneA_id = "${module.vpc.private_subnet_zoneA_id}"
}
```
./modules/ec2/main.tf
```hcl
resource "aws_instance" "private_subnet_win_server" {
    ami = "${lookup(var.aws_amis,var.region)}"
    instance_type = "${var.ec2_type}"
    subnet_id = "${var.private_subnet_zoneA_id}"[0]
    key_name        = "${var.ssh_key_name}"

    root_block_device {
        volume_type = "gp2"
        volume_size = "30"
        encrypted = true
        kms_key_id = "some-id"
        // tags = {
        //     Name = "some-name"
        // }
    }
    
    tags = {
        Name = "some-name"
    }
}
```
### Expected Behavior
Terraform run will spin up the ec2 instance with an encrypted root volume.

### Actual Behavior
```
Error: Unsupported argument

  on modules/ec2/main.tf line 22, in resource "aws_instance" "private_subnet_win_server":
  22:         encrypted = true

An argument named "encrypted" is not expected here.


Error: Unsupported argument

  on modules/ec2/main.tf line 23, in resource "aws_instance" "private_subnet_win_server":
  23:         kms_key_id = "xxxxxx-xxxxxx-xxxxx-xxxxxx"

An argument named "kms_key_id" is not expected here.
```

### Steps to Reproduce

1. Use resource "aws_instance" create an ec2 instance and put a "root_block_device" mapping block inside of that resource like I did
2. Run "terraform plan"
3. Error

### References

* #0000

**Comment by alexistimic** [comment]
Same issue on the `launch_configuration` root_block_device

**Comment by bflad** [comment]
Hi folks 👋 As mentioned above, this functionality was previously released in [version 2.23.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2230-august-07-2019) and has been available in all releases since. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

The Terraform AWS Provider documentation website currently only shows the latest version (and its functionality), however we have plans to support versioned provider documentation 🔜 .

**Comment by krazik** [comment]
This issue seems to be back in 2.28, but is ok in 2.27
